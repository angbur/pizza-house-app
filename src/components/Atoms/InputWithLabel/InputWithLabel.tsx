import { ThemeContext } from 'components/Theme/ThemeContext';
import { ComponentPropsWithoutRef, useContext } from 'react';
import styled from 'styled-components';

const Box = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 0.625rem;
    & > label {
        color: ${(props) => props.theme?.palette.light};
    }
    & > input {
        background: ${(props) => props.theme?.palette.light};
        &:focus-visible {
            outline: none;
        };
        padding: 0.625rem;
        border-radius: 8px;
        border: none;
        width: 300px;
    }
`;

type LabelProps = {
    label: string;
};

const InputWithLabel = (props: ComponentPropsWithoutRef<'input'> & LabelProps) => {
    const theme = useContext(ThemeContext);
    return (
        <Box theme={theme}>
            <label htmlFor={props.id}>
                {props.label}
            </label>
            <input id={props.id} placeholder={props.placeholder}/>
        </Box>
    )
};

export default InputWithLabel;