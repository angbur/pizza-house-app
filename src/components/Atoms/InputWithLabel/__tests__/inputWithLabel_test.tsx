import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import InputWithLabel from '../InputWithLabel';

const name = 'Bernard';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <InputWithLabel
        label={'Name'}
        name={'firstName'}
        type={'text'}
        placeholder={'Your first name'}
        color={'light'}
        value={name}
        onChange={jest.fn()}
        error={!/^[A-Za-zżźćńółęąśŻŹĆĄŚĘŁÓŃ]{2,50}$/.test(name.trim())}
      />
    </ThemeProvider>
  );
};

test('renders correctly input with label', () => {
  const tree = renderer.create(<Wrapper />).toJSON();
  expect(tree).toMatchSnapshot();
});
