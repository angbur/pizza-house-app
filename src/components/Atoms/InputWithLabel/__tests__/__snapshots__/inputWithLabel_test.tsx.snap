// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly input with label 1`] = `
<div
  className="sc-bcXHqe dHZfIZ"
>
  <label
    style={
      Object {
        "color": "#FCFBFB",
      }
    }
  >
    Name
  </label>
  <input
    name="firstName"
    onChange={[MockFunction]}
    placeholder="Your first name"
    style={
      Object {
        "color": "#151515",
      }
    }
    type="text"
    value="Bernard"
  />
  <p
    className="paragraph"
    style={
      Object {
        "color": "#e74c3c",
        "fontFamily": "Source Sans Pro",
        "fontSize": "1rem",
        "fontWeight": "400",
        "letterSpacing": "0",
        "lineHeight": "1.6",
      }
    }
  />
</div>
`;
